pkg/image/apis/image/validation/validation.go

```diff
@@ -87,13 +87,6 @@ func ValidateImageSignature(signature *imageapi.ImageSignature) field.ErrorList
 
 func validateImageSignature(signature *imageapi.ImageSignature, fldPath *field.Path) field.ErrorList {
 	allErrs := validation.ValidateObjectMeta(&signature.ObjectMeta, false, path.ValidatePathSegmentName, fldPath.Child("metadata"))
-	if len(signature.Labels) > 0 {
-		allErrs = append(allErrs, field.Forbidden(fldPath.Child("metadata").Child("labels"), "signature labels cannot be set"))
-	}
-	if len(signature.Annotations) > 0 {
```

Comment:
The decision to avoid these was primarily based on the idea that labels and annotations that cannot be "normally" modified are not terribly useful and break CLI usage.  Nested annotations and labels caused problems long ago on imagestreamtags and my fix from way back when didn't even cover all cases (found that out recently).

I wanted to avoid the problem, but this bug https://github.com/openshift/origin/pull/19030 effectively allowed this to be bypassed one time.  Now we have a mess.

Reply:
right, well in the meantime the imagesignature controller was apparently written to expect that it would be updating/adding these annotations.  So was it designed wrong, or do we need to permit this annotations to be added/updated?

Reply:
> right, well in the meantime the imagesignature controller was apparently written to expect that it would be updating/adding these annotations. So was it designed wrong, or do we need to permit this annotations to be added/updated?

I would strip that out in the PrepareForUpdate unless @mfojtik has a strong reason for needing it.

Reply:
presumably we should also remove the logic from the controller that is currently adding this annotation.